plugins/database/mongodb/connection_producer.go

```diff
@@ -42,6 +46,29 @@ func (c *mongoDBConnectionProducer) Initialize(conf map[string]interface{}, veri
 		return fmt.Errorf("connection_url cannot be empty")
 	}
 
+	if c.WriteConcern != "" {
+		input := c.WriteConcern
+
+		// Try to base64 decode the input. If successful, consider the decoded
+		// value as input.
+		inputBytes, err := base64.StdEncoding.DecodeString(input)
+		if err == nil {
+			input = string(inputBytes)
+		}
+
+		concern := &mgo.Safe{}
+		err = json.Unmarshal([]byte(input), concern)
+		if err != nil {
+			return fmt.Errorf("error mashalling write concern: %s", err)
+		}
+
+		// Only set c.safe if anything got marshaled into concern. We don't want to
+		// pass an empty, non-nil mgo.Safe object into mgo.SetSafe in Connection().
+		if (mgo.Safe{} != *concern) {
+			c.safe = concern
```

Comment:
Perhaps this should be an error case? If they set the WriteConcern JSON we should expect to have successfully marshaled something